yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -656,23 +661,40 @@ func (yb *TargetYugabyteDB) importBatchFastRecover(conn *pgx.Conn, batch Batch,
 			}
 		}
 
-		fields := strings.Split(line, args.Delimiter)
-		stmtArgs := make([]interface{}, len(fields))
-		for i, val := range fields {
-			stmtArgs[i] = val
+		var singleLineReader io.Reader
+		if copyHeader != "" {
+			singleLineReader = strings.NewReader(copyHeader + "\n" + line)
+		} else {
+			singleLineReader = strings.NewReader(line)
 		}
 
-		// 5. Execute the INSERT statement one by one
-		res, err := conn.Exec(context.Background(), insertStmt, stmtArgs...)
+		// 5. Execute the COPY statement with the line read from the file
+		res, err := conn.PgConn().CopyFrom(context.Background(), singleLineReader, copyCommand)
 		if err != nil {
-			return 0, fmt.Errorf("insert into target table: %w", err)
+			// Ignore err if VIOLATES_UNIQUE_CONSTRAINT_ERROR only
+			if strings.Contains(err.Error(), VIOLATES_UNIQUE_CONSTRAINT_ERROR) {
+				// logging lineNum might not be useful as batches are truncated later on
+				log.Debugf("ignoring error %q for line=%q in batch %q", err.Error(), line, batch.GetFilePath())
+				rowsIgnored++ // increment before continuing to next line
+				continue
+			}
+
+			var pgerr *pgconn.PgError
+			if errors.As(err, &pgerr) {
+				err = fmt.Errorf("%s, %s in %s", err.Error(), pgerr.Where, batch.GetFilePath())
+			}
```

Comment:
what is the purpose of this block?

Reply:
i think this just adds more context to the error message.
Copied it from the copyBatchCore() function